### ISSUE TITLE
Transformer v2 fails when converting a table with an index on a primaryKey

### ISSUE THREAD

**Issue by alexhouse** [aws-amplify/amplify-cli#9292]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm -g

### If applicable, what version of Node.js are you using?

14.18.1

### Amplify CLI Version

7.6.3

### What operating system are you using?

OSX 12

### Amplify Categories

api

### Amplify Commands

push

### Describe the bug

I'm attempting to migrate to Transformer Version 2 (`amplify api migrate`) in order to perform some other GraphQL updates via `amplify push`.

Once the new schema has been generated, any `amplify push` (on a new environment) end up breaking with the following error:

```
✖ An error occurred when pushing the resources to the cloud
🛑 Could not fetch keySchema for Model.
InvalidDirectiveError: Could not fetch keySchema for Model.
    at AuthTransformer.protectListResolver (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/graphql-auth-transformer/src/graphql-auth-transformer.ts:499:13)
    at AuthTransformer.generateResolvers (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/graphql-auth-transformer/src/graphql-auth-transformer.ts:326:16)
    at GraphQLTransform.transform (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/graphql-transformer-core/src/transformation/transform.ts:252:21)
    at _buildProject (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/graphql-transformer/transform-graphql-schema.ts:564:37)
    at buildAPIProject (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/graphql-transformer/transform-graphql-schema.ts:521:24)
    at transformGraphQLSchema (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/graphql-transformer/transform-graphql-schema.ts:394:29)
    at Object.run (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/push-resources.ts:166:5)
    at async Promise.all (index 0)
    at providersPush (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/src/extensions/amplify-helpers/push-resources.ts:139:3)
    at AmplifyToolkit.pushResources [as _pushResources] (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/src/extensions/amplify-helpers/push-resources.ts:96:9)
    at Object.executeAmplifyCommand (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-api/src/index.ts:222:5)
    at executePluginModuleCommand (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:204:3)
    at executeCommand (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:30:5)
    at Object.run (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/src/index.ts:205:5)
```

Digging into the amplify cli code a bit and performing various console debugs, it looks like an `@index` has been created on the `@primaryKey` in the schema, but the `@index` gets ignored.

Original schema:
```graphql
type Model
@model
@auth(rules: [
  {allow: public, provider: apiKey}
])
@key(name: "byUser", fields: ["user", "channel"], queryField: "listModelsByUser")
@key(name: "byChannel", fields: ["channel", "user"], queryField: "listModelsByChannel")
@key(fields: ["user", "channel", "token", "secret"])
{
  user: String!
  channel: String!
  token: String!
  secret: String
}
```

Transformed schema:

```graphql
type Model @model @auth(rules: [{allow: public, provider: apiKey}]) {
  user: String! @primaryKey(sortKeyFields: ["channel", "token", "secret"]) @index(name: "byUser", queryField: "listModelsByUser", sortKeyFields: ["channel"])
  channel: String! @index(name: "byChannel", queryField: "listModelsByChannel", sortKeyFields: ["user"])
  token: String!
  secret: String!
}
```

The `table.globalSecondaryIndexes` only contains the `byChannel` index and throws the InvalidDirect exception.

### Expected behavior

`amplify push` should succeed, finding all available indexes

### Reproduction steps

Create the following table using v1 transformer:

```graphql
type Model
@model
@auth(rules: [
  {allow: public, provider: apiKey}
])
@key(name: "byUser", fields: ["user", "channel"], queryField: "listModelsByUser")
@key(name: "byChannel", fields: ["channel", "user"], queryField: "listModelsByChannel")
@key(fields: ["user", "channel", "token", "secret"])
{
  user: String!
  channel: String!
  token: String!
  secret: String
}
```

Run `amplify push`
Run `amplify api migrate`
Run `amplify push`

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line

type Model
@model
@auth(rules: [
  {allow: public, provider: apiKey}
])
@key(name: "byUser", fields: ["user", "channel"], queryField: "listModelsByUser")
@key(name: "byChannel", fields: ["channel", "user"], queryField: "listModelsByChannel")
@key(fields: ["user", "channel", "token", "secret"])
{
  user: String!
  channel: String!
  token: String!
  secret: String
}

```

</details>

### Log output

<details>

```
# Put your logs below this line

✖ An error occurred when pushing the resources to the cloud
🛑 Could not fetch keySchema for Model.
InvalidDirectiveError: Could not fetch keySchema for Model.
    at AuthTransformer.protectListResolver (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/graphql-auth-transformer/src/graphql-auth-transformer.ts:499:13)
    at AuthTransformer.generateResolvers (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/graphql-auth-transformer/src/graphql-auth-transformer.ts:326:16)
    at GraphQLTransform.transform (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/graphql-transformer-core/src/transformation/transform.ts:252:21)
    at _buildProject (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/graphql-transformer/transform-graphql-schema.ts:564:37)
    at buildAPIProject (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/graphql-transformer/transform-graphql-schema.ts:521:24)
    at transformGraphQLSchema (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/graphql-transformer/transform-graphql-schema.ts:394:29)
    at Object.run (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/push-resources.ts:166:5)
    at async Promise.all (index 0)
    at providersPush (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/src/extensions/amplify-helpers/push-resources.ts:139:3)
    at AmplifyToolkit.pushResources [as _pushResources] (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/src/extensions/amplify-helpers/push-resources.ts:96:9)
    at Object.executeAmplifyCommand (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-api/src/index.ts:222:5)
    at executePluginModuleCommand (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:204:3)
    at executeCommand (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:30:5)
    at Object.run (/Users/.nvm/versions/node/v14.18.1/lib/node_modules/@aws-amplify/cli/src/index.ts:205:5)

```

</details>

### Additional information

Help me Obi Wan Kenobi ⚔️

**Comment by cjihrig** [comment]
@alexhouse I'm not able to reproduce the issue. First, I created a new project and deployed the v2 schema you provided. That went through fine.

Next, I created another new project using the v1 schema you provided. There are a couple issues with the provided schema. There is no `deviceToken` or `expoPushToken` field in the schema. For the v1 primary key, I used the fields `user`, `channel`, and `token`, since the provided schema was unclear. Once I corrected those errors, I was able to migrate and push the schema just fine.

**Comment by alexhouse** [comment]
Hmm.
In my attempt to provide a simple table to reproduce the issue I messed up. I've edited the OP with the correct schemas.

I've just noticed that the `byUser` key is currently labelled as a local secondary instead of a global secondary within DynamoDB... that may well be the cause of the errors. I have a feeling that the only way of making a local key global is to fully recreate the table, right? Which would then require all data to be upserted somehow (batch operations?)
